core/src/main/java/com/sequenceiq/cloudbreak/core/cluster/ClusterManagerUpscaleService.java

```diff
@@ -86,16 +87,26 @@ public void upscaleClusterManager(Long stackId, Map<String, Integer> hostGroupWi
     }
 
     private void handleWaitForHostsTimeout(boolean repair, boolean primaryGatewayChanged, StackView stack, ExtendedPollingResult result) {
-        String message = String.format("Upscaling cluster manager were not successful, waiting for hosts timed out for nodes: %s",
-                result.getFailedInstanceIds());
-        LOGGER.warn(message);
-        if (!repair && !primaryGatewayChanged && targetedUpscaleSupportService.targetedUpscaleOperationSupported(stack)) {
-            instanceMetaDataService.updateInstanceStatuses(result.getFailedInstanceIds(), InstanceStatus.ZOMBIE,
-                    "Upscaling cluster manager were not successful, waiting for hosts timed out");
+        if (CollectionUtils.isNotEmpty(result.getFailedInstanceIds())) {
+            String message = String.format("Upscaling cluster manager were not successful, waiting for hosts timed out for nodes: %s",
+                    result.getFailedInstanceIds());
+            LOGGER.warn(message);
+            if (!repair && !primaryGatewayChanged && targetedUpscaleSupportService.targetedUpscaleOperationSupported(stack)) {
+                instanceMetaDataService.updateInstanceStatuses(result.getFailedInstanceIds(), InstanceStatus.ZOMBIE,
+                        "Upscaling cluster manager were not successful, waiting for hosts timed out");
+            } else {
+                instanceMetaDataService.updateInstanceStatuses(result.getFailedInstanceIds(), InstanceStatus.ORCHESTRATION_FAILED,
+                        "Upscaling cluster manager were not successful, waiting for hosts timed out");
+                throw new CloudbreakServiceException(message);
+            }
+        } else if (result.getException() != null) {
+            // if getFailedInstanceIds is null, then possibly something is wrong with CM, so we should throw exception anyway
+            LOGGER.error("Upscaling cluster manager were not successful, waiting for hosts timed out because: ", result.getException());
+            throw new CloudbreakServiceException(result.getException());
         } else {
-            instanceMetaDataService.updateInstanceStatuses(result.getFailedInstanceIds(), InstanceStatus.ORCHESTRATION_FAILED,
-                    "Upscaling cluster manager were not successful, waiting for hosts timed out");
-            throw new CloudbreakServiceException(message);
+            LOGGER.error("Upscaling cluster manager were not successful, waiting for hosts timed out!");
+            throw new CloudbreakServiceException("Upscaling cluster manager were not successful, waiting for hosts timed out, " +
+                    "please check Cloudera Manager logs fur further details.");
```

Comment:
I am not sure about these `else` branches. 
These instances will remain in SERVICES_RUNNING status and I am not sure if they will be handled properly later.  Have you tested this scenario?

Reply:
the same happening when failedInstanceIds is empty,here the task was to not send null but some proper error message,handling instance status in these cases would be another task

Reply:
I have created a new task for handling the scenario above.

Reply:
thanks!